cmd/gapit/server_performance.go

```diff
@@ -0,0 +1,89 @@
+// Copyright (C) 2017 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"context"
+	"flag"
+	"time"
+
+	"github.com/google/gapid/core/app"
+	"github.com/google/gapid/core/log"
+	"github.com/google/gapid/gapis/service"
+	gapidPath "github.com/google/gapid/gapis/service/path"
+)
+
+type serverPerformanceVerb struct{ ServerPerformanceFlags }
+
+func init() {
+	verb := &serverPerformanceVerb{
+		ServerPerformanceFlags{LoopCount: 1},
+	}
+	app.AddVerb(&app.Verb{
+		Name:      "server_performance",
+		ShortHelp: "Test the performance of GAPIS.",
+		Action:    verb,
+	})
+}
+
+func (verb *serverPerformanceVerb) Run(ctx context.Context, flags flag.FlagSet) error {
+
+	if flags.NArg() != 1 {
+		app.Usage(ctx, "Exactly one gfx trace file expected, got %d", flags.NArg())
+		return nil
+	}
+
+	client, capturePath, err := getGapisAndLoadCapture(ctx, verb.Gapis, verb.Gapir, flags.Arg(0), verb.CaptureFileFlags)
+	if err != nil {
+		return err
+	}
+	defer client.Close()
+
+	var device *gapidPath.Device
+	if !verb.OriginalDevice {
+		device, err = getDevice(ctx, client, capturePath, verb.Gapir)
+		if err != nil {
+			return err
+		}
+	}
+
+	opts := &service.ExportReplayOptions{
+		FramebufferAttachments: make([]*gapidPath.FramebufferAttachment, 0),
+		GetTimestampsRequest:   nil,
+		DisplayToSurface:       false,
+		LoopCount:              int32(verb.LoopCount),
+	}
+
+	start := time.Now()
```

Comment:
That's good for a first version, however we're measuring time w/o taking into account the conditions in which we're running this test. A future version of this benchmark could inform and/or normalise timings based on e.g. num CPU cores, occupation, etc Some of these values may not be accessible by this command but we can at least warn the user about it.

Reply:
Yeah, I imagine we'll add more as time goes on. I'm not sure we can really normalise for everything though. Computers are so different in their performance.